lib/components/map/connected-transitive-overlay.js

```diff
@@ -1,44 +1,71 @@
+import isEqual from 'lodash.isequal'
 import coreUtils from '@opentripplanner/core-utils'
 import TransitiveCanvasOverlay from '@opentripplanner/transitive-overlay'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
-import { getActiveSearch, getActiveItinerary, getActiveItineraries } from '../../util/state'
+import { ComponentContext } from '../../util/contexts'
+import { getActiveItinerary, getActiveItineraries, getActiveSearch } from '../../util/state'
+
+/**
+ * Wrapper for TransitiveCanvasOverlay that passes getTransitiveRouteLabel defined in ComponentContext.
+ */
+class TransitiveCanvasOverlayWithContext extends Component {
+  static contextType = ComponentContext
+
+  shouldComponentUpdate (nextProps) {
+    return (
+      !isEqual(nextProps.activeSearchVisibleItinerary, this.props.activeSearchVisibleItinerary) ||
+      !isEqual(nextProps.activeSearchResponseOtp, this.props.activeSearchResponseOtp)
+    )
+  }
+
+  render () {
+    const { activeSearchResponseOtp, activeSearchVisibleItinerary, ...otherProps } = this.props
+    let transitiveData = null
+    // TODO: prevent itineraryToTransitive() from being called more than needed
+    //       (partially addressed by shouldComponentUpdate)
+    if (activeSearchVisibleItinerary) {
+      transitiveData = coreUtils.map.itineraryToTransitive(activeSearchVisibleItinerary, null, this.context.getTransitiveRouteLabel)
+    } else if (activeSearchResponseOtp) {
+      transitiveData = activeSearchResponseOtp
+    }
```

Comment:
It might look cleaner to move all this into the mapStateToProps since in the end only the transitiveData variable is needed in the actual stuff that gets rendered..